R/mlflow/R/model.R

```diff
@@ -0,0 +1,167 @@
+#' Save Model for MLflow
+#'
+#' Saves model in MLflow's format that can later be used
+#' for prediction and serving.
+#'
+#' @param x The serving function or model that will perform a prediction.
+#' @param path Destination path where this MLflow compatible model
+#'   will be saved.
+#' @param dependencies Optional vector of paths to dependency files
+#'   to include in the model, as in \code{r-dependencies.txt}
+#'   or \code{conda.yaml}.
+#'
+#' @importFrom yaml write_yaml
+#' @export
+mlflow_save_model <- function(x, path = "model", dependencies = NULL) {
```

Comment:
Well, I understand the suggestion... However, each model would also be different. We could generalize this function to save a list of models, something like:

```
tf_graph <- train_tensorflow_model()
predictor <- crate(~ stats::predict(model, .x), model)

# Save model to predict in GPU or CPU
mlflow_save_model(c(tf_graph, predictor), path = "model")
```

Which feels a bit far fetched since we don't even have an implementation for TF, etc. The good news, we can implement this later without breaking the API compatibility since everything in R is a vector.